File: tests/benchmarks/list/list.py
from time import process_time as clock
from math import floor

def reverse_list(list):
	count = floor(len(list) / 2)
	for i in range(count):
		last_index = len(list) - i - 1
		list[i], list[last_index] = list[last_index], list[i]
	return list

N = 20000000
l = list(range(N))

start = clock()
reverse_list(l)
print('elapsed: ', clock() - start, 's')

File: tests/benchmarks/list/list.rb

def reverse_list(list)
  (list.length >> 1).times do |i|
    last_index = list.length - i - 1
    list[i], list[last_index] = list[last_index], list[i]
  end
end

N = 20000000
list = (0...N).to_a

start = Time.now
reverse_list(list)
puts "elapsed: " + (Time.now - start).to_s + ' s'
